.github/workflows/python-package.yml

```diff
@@ -0,0 +1,233 @@
+name: Build Python Package
+
+on:
+  push:
+    branches:
+      - add-pypi-package
+
+jobs:
+  sdist:
+    name: Build the sdist
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install libboost-dev
+    - uses: actions/checkout@v2
+      name: Checkout the repository
+      with:
+        submodules: recursive
+    - name: Set Up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+    - name: Update pip
+      run: python3 -m pip install -U pip setuptools
+    - name: Install dependencies
+      run: python3 -m pip install scons numpy cython build
+    - name: Build the interface
+      run: python3 `which scons` build f90_interface=n debug=n googletest=none python_package=sdist use_pch=n
```

Comment:
Oh, it's because all the source files get copied in to the `sdist` in this build step, so the PCH doesn't actually ever get used, since no compiling is done at this stage.